src/main/resources/db/migration/migrate_data_from_director_v1.sql

```diff
@@ -0,0 +1,155 @@
+INSERT director_v2.repo_namespaces
+SELECT * FROM director.repo_names v1
+ON DUPLICATE KEY UPDATE repo_id = v1.repo_id, created_at = v1.created_at, updated_at = v1.updated_at;
+
+-- select count(*) from repo_namespaces ;
+
+DROP TEMPORARY TABLE IF EXISTS ecu_targets_v1;
+create temporary table ecu_targets_v1 AS
+select namespace, max(id) id, filepath filename, length, max(checksum) checksum, hash sha256, uri, max(created_at) created_at, max(updated_at) updated_at from (
+select namespace, uuid() id, filepath, length, checksum, json_unquote(json_extract(checksum, '$.hash')) hash, NULL uri, created_at, updated_at
+from director.current_images c
+UNION
+select namespace, uuid(), target filepath, target_size length, JSON_OBJECT("method", "sha256", "hash", target_hash), target_hash, target_uri uri, created_at, updated_at
+from director.multi_target_updates mtu
+UNION
+select namespace, uuid(), from_target filepath, from_target_size length, JSON_OBJECT("method", "sha256", "hash", from_target_hash), from_target_hash, from_target_uri uri, created_at, updated_at
+from director.multi_target_updates mtu
+where
+ from_target is not null AND from_target_hash is not null
+) i group by namespace, filename, hash, uri, length -- if uri, length needs to be here, CREATE UNIQUE INDEX will fail and we need to check why
+;
+
+create unique index if not exists ecu_targets_v1_uniq_idx on ecu_targets_v1 (namespace, `filename`(500), sha256);
+create unique index if not exists ecu_targets_v1_id_uniq_idx on ecu_targets_v1 (id);
+
+update ecu_targets v2 join ecu_targets_v1 v1 using (namespace, filename, sha256) SET
+   v1.id = v2.id,
+   v2.uri = v1.uri,
+   v2.length = v1.length
+   ;
+
+-- ecu_targets
+INSERT INTO ecu_targets (namespace, id, filename, length, checksum, sha256, uri, created_at, updated_at)
+select namespace, id, filename, length, checksum, sha256, uri, created_at, updated_at from ecu_targets_v1
+WHERE (namespace, filename, sha256) NOT IN (select namespace, filename, sha256 from ecu_targets)
+;
+
+--select count(*) FROM (SELECT * from (
+--select namespace, filepath, json_unquote(json_extract(checksum, '$.hash')) checksum, length from director.current_images
+--UNION
+--select namespace, target, target_hash, target_size from director.multi_target_updates
+--UNION
+--select namespace, from_target, from_target_hash, from_target_size from director.multi_target_updates  WHERE from_target is not null and from_target_hash is not null
+--) _t1 GROUP by 1, 2, 3, 4) _t2
+--;
+
+-- ecus
+insert into director_v2.ecus (namespace, ecu_serial, device_id, public_key, hardware_identifier, current_target, created_at, updated_at)
+select e.namespace, e.ecu_serial, device, public_key, hardware_identifier, et.id, e.created_at, e.updated_at
+FROM director.ecus e
+LEFT JOIN (select c.ecu_serial, t.id, t.namespace, t.filename, t.checksum FROM director.current_images c JOIN ecu_targets_v1 t ON t.namespace = c.namespace AND t.filename = c.filepath AND t.checksum = c.checksum) et
```

Comment:
So this also copies over ECUs which are not in `director.current_images`, right? Makes sense.

Reply:
yes